string filename   # Name of bag file to save snapshot to. 
                  # if it ends in .bag, this exact filename will be saved.
                  # otherwise, the current datatime and .bag will be appended to the name
                  #   example: "test" -> test2018-05-23-10-04-20.bag
                  #   example: "test.bag" -> test.bag
string[] topics   # List of topics to include in snapshot. 
                  # If empty, all buffered topics will be written
time start_time   # Earliest timestamp for a message to be included in written bag
time stop_time    # Latest timestamp for a message to be incldued in written bag

---

bool success    # True if snapshot succesfully written to disk

string NO_DATA=no messages buffered on selected topics
string message  # Error description if failed
